src/components/dropdown/Dropdown.vue

```diff
@@ -188,11 +201,23 @@ export default {
         * Close dropdown if clicked outside.
         */
         clickedOutside(event) {
+            if (this.cancelOptions.indexOf('outside') < 0) return
             if (this.inline) return
 
             if (!this.isInWhiteList(event.target)) this.isActive = false
         },
 
+        /**
+         * Keypress event that is bound to the document
+         */
+        keyPress(event) {
+            // Esc key
+            if (this.isActive && event.keyCode === 27) {
+                if (this.cancelOptions.indexOf('escape') < 0) return
```

Comment:
Maybe we should move it to function start?

Reply:
Thanks for your comment. Not sure about the order.

Since `keyup` is bound to the document, every key pressed in the page will trigger this function. If we fill a form or something, I want to exit the function as soon as possible if it is not needed. In this case, if the dropdown is not active, it wont even need to check whick key were pressed.

What are your thoughts on this?

Reply:
How do you think, regular user, push esc on page too often?) xD

Reply:
I'm not saying users press esc too often. I'm just saying that I'm not sure about the benefit of switching the conditions. Surely wont be that huge of an optimization.

Reply:
But it's optimization.

2 ways
closed dropdown
opened dropdown

1: pushing a - call func check active -> exit
    write some another text on page - check active -> exit

2: pushing a - call func -> exit
    pushing another text in some field for example ->exit

Idk, that's your choice, ok.

Reply:
For me it's ok now.
@afwn90cj93201nixr2e1re Maybe you're right but I don't think about huge  decrease of performance.
If you have time you might make a benchmark ;)